> Checking for code style violations in OpenRA.Mods.Common...
OpenRA.Mods.Common/Widgets/Logic/ColorPickerLogic.cs:L28: [SP2100] Code line must be no longer than 180 characters (now 189).
OpenRA.Mods.Common/Widgets/Logic/ColorPickerLogic.cs:L31: [SP2100] Code line must be no longer than 180 characters (now 187).
make: *** [check] Error 1

--- a/OpenRA.Mods.Common/Widgets/Logic/ColorPickerLogic.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/ColorPickerLogic.cs
@@ -25,11 +25,12 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 		int paletteTabHighlighted = 0;
 
 		[ObjectCreator.UseCtor]
-		public ColorPickerLogic(Widget widget, ModData modData, World world, HSLColor initialColor, Action<HSLColor> onChange, Dictionary<string, MiniYaml> logicArgs)
+		public ColorPickerLogic(Widget widget, ModData modData, World world, HSLColor initialColor, string initialFaction, Action<HSLColor> onChange,
+			Dictionary<string, MiniYaml> logicArgs)
 		{
 			string actorType;
-			if (!ChromeMetrics.TryGet("ColorPickerActorType", out actorType))
-				actorType = "mcv";
+			if (initialFaction == null || !ChromeMetrics.TryGet("ColorPickerActorType-" + initialFaction, out actorType))
+				actorType = ChromeMetrics.Get<string>("ColorPickerActorType");
 
 			var preview = widget.GetOrNull<ActorPreviewWidget>("PREVIEW");
 			var actor = world.Map.Rules.Actors[actorType];
@@ -194,7 +195,8 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 			var colorChooser = Game.LoadWidget(world, "COLOR_CHOOSER", null, new WidgetArgs()
 			{
 				{ "onChange", onChange },
-				{ "initialColor", Game.Settings.Player.Color }
+				{ "initialColor", Game.Settings.Player.Color },
+				{ "initialFaction", null }
 			});
 
 			color.AttachPanel(colorChooser, onExit);

--- a/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs
@@ -217,7 +217,8 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 			var colorChooser = Game.LoadWidget(world, "COLOR_CHOOSER", null, new WidgetArgs()
 			{
 				{ "onChange", onChange },
-				{ "initialColor", client.Color }
+				{ "initialColor", client.Color },
+				{ "initialFaction", client.Faction }
 			});
 
 			color.AttachPanel(colorChooser, onExit);

